sound/soc/sof/intel/hda-stream.c

```diff
@@ -570,14 +587,18 @@ static bool hda_dsp_stream_check(struct hdac_bus *bus, u32 status)
 			snd_hdac_stream_writeb(s, SD_STS, sd_status);
 
 			active = true;
-			if (!s->substream ||
+			if (!(s->substream || s->cstream) ||
```

Comment:
@crojewsk this check is confusing me. is the intention to check if both s->substream and s->cstream are NULL?

Reply:
@ranj063 Take a look at initial state:
`if (!s-> substream || <more checks here>)`
What could simply be translated into human language as:
_check if HDA stream is setup and running before executing following instructions_

As in previous case, it was always pcm (substream) being tied to hdac stream. Compr-support changes that state, and thus relevant checks are updated to:
_check if pcm OR compr stream is up and running_

If both are NULL, than statement evaluates to TRUE.
If both are not-NULL, than statement evaluates to FALSE.
If either is not-NULL, than statement evaluates to FALSE.

FALSE in above means: let's continue with _elapsed operation.
TRUE in above means: let's skip this stream and check the next one.

Reply:
in other patches you wrote this differently, e.g. 			if (!s->substream && !s->cstream))

While it's equivalent, following the same patterns helps reviewers

Reply:
No problem. Reverted to my old solution so it is not confusing reviewer, in v6.